### DISCUSSION TITLE
I'm trying to figure out how to inject custom blog post frontmatter into the HTML

### DISCUSSION THREAD

**Discussion by divSelector** [squidfunk/mkdocs-material#6957] [Q&A] [answered]
I have this in the frontmatter of my post

```yml
divchan_board: main
divchan_thread: 0x000860
divchan_post: 0x000001
```

I have this in `blog-post.html`

```html
    <!-- Values Passed From Blog Post Frontmatter to JS -->
    <input type="hidden" id="divchan-board" value="{{ page.divchan_board }}">
    <input type="hidden" id="divchan-thread" value="{{ page.divchan_thread }}">
    <input type="hidden" id="divchan-post" value="{{ page.divchan_post }}">
```

It's successfully putting the input elements into the blog posts, but the values are empty. I'm not sure if there is a way to go about this or if I need to write some kind of plugin. If someone could point me in the right direction I can figure it out.

**Comment by alexvoss** [comment, marked as answer]
Try adding `.meta` into the variable reference, as in:

```html
<input type="hidden" id="divchan-board" value="{{ page.meta.divchan_board }}">
```

**Reply by divSelector** [reply]
Oh, wow. I thought you would be telling me how to write a context processor :smile: 

thanks
